qa/test_template.sh

```diff
@@ -5,6 +5,9 @@ set -e
 # Force tests to be verbose
 set -x
 
+# launch the heartbeat  so we can see if the test hang or just is slow
+counter=0; while [ 1 ]; do echo "Heartbeat  $counter"; counter=$((counter+1)); sleep 10; done &
```

Comment:
Also, this will interfere with test output - making it hard to search, for instance.

Reply:
That is the idea. If the heartbeat interleaves with logs that is okay. If the test hang we see only it in the log.

Reply:
> Also, this will interfere with test output - making it hard to search, for instance.

If it works then you don't have to look for it, if it hangs you will see a plenty of it in the log tail.

Reply:
> If it works then you don't have to look for it, if it hangs you will see a plenty of it in the log tail.

If there is a failure reported, you will get the numbers interleaved in the error message in the worst case.

> That is the idea. If the heartbeat interleaves with logs that is okay. If the test hang we see only it in the log.

But how do you differentiate between test that runs long vs one that hanged? And at the end of the job you get a Timeout error either way.

Maybe we should print the timestamps in the tests, they are available in the even: https://docs.nose2.io/en/latest/dev/event_reference.html#nose2.events.StartTestEvent

Or maybe the plugin used for timing already has this option?

Reply:
> But how do you differentiate between test that runs long vs one that hanged? And at the end of the job you get a Timeout error either way.

I guess the whole test is slow and you get a lot of beats between the logs as the test progresses. If it hangs the beats are rare and then it accumulates at the end.

I'm not saying it is a silver bullet, my idea was to give it a go and check if it helps or if we should revert as not useful.

The beat frequency for TL0_self-test:
![image](https://user-images.githubusercontent.com/39967756/226632920-4b11d199-297a-45d4-905c-f1377d5a5794.png)

TL0_python-self-test-operators_1:
![image](https://user-images.githubusercontent.com/39967756/226633448-bd13c1c3-222f-4e58-9d91-a24cfc3f87d6.png)

Slow regions are dense.